src/devtools/client/debugger/src/components/PrimaryPanes/SourcesTree.tsx

```diff
@@ -94,21 +89,33 @@ interface STState {
 
 class SourcesTree extends Component<PropsFromRedux, STState> {
   constructor(props: PropsFromRedux) {
-    super(props);
-    const { sources } = this.props;
+    const updatedProps = {
+      ...props,
+      cx: { navigateCounter: 0 },
+      sourcesLoading: false,
+      shownSource: null,
+      selectedSource: null,
+      focused: null,
+      sources: {},
+      sourceCount: 0,
+    };
```

Comment:
Can you explain what this is doing and why it's necessary? `props` should always come from _outside_ of a component. It's okay to set _default_ values, but this is actually overriding any external values with hard-coded values (that won't ever change) so this seems like a bug for sure.